Profile documentation (#9973)
kyma-project/kyma: docs/kyma/05-05-profiles.md

---
title: Profiles
type: Configuration
---

The Kyma Operator allows you to use pre-defined profiles to install or upgrade Kyma. To install Kyma using a profile, you must specify the **spec.profile** attribute in the [Installation custom resource](#custom-resource-installation) (CR).

Profile is a set of helm values defined in `profile-{name}.yaml` file at the component chart folder. If the profile file is not present, the Kyma Operator will use default set of values from the `values.yaml` file. The following fragment of the `values.yaml` shows definition limitRange settings:

```yaml
limitRange:
  max:
    memory: 4Gi
  default:
    memory: 96Mi
  defaultRequest:
    memory: 32Mi
```

To override, for example `limitRange.max.memory` put following in the `profile-{name}.yaml` file: 

```yaml
limitRange:
  max:
    memory: 5Gi
```

>**NOTE:** If the `profile-{name}.yaml` file doesn't exist in a component chart simply create it.

> On top of the applied profile the Kyma Operator will put user-defined overrides. For more information, see the [overrides section](#configuration-helm-overrides-for-kyma-installation).

Author: there's no possibility to define custom profiles, only evaluation & production is possible

> On top of the applied profile the Kyma Operator will put user-defined overrides. For more information, see the [overrides section](#configuration-helm-overrides-for-kyma-installation).

Reviewer: I missed some important code, that's true, thanks

---
title: Profiles
type: Configuration
---

> The Kyma Operator allows you to use pre-defined profiles to install or upgrade Kyma. To install Kyma using a profile, you must specify the **spec.profile** attribute in the [Installation custom resource](#custom-resource-installation) (CR).

Reviewer 2: I'd add general info about profiles in the 2nd sentence and move `To install Kyma using a profile...` part to the next paragraph (or even after the paragraph explaining values overrides, we'll see 🤔 ).

> The profile is a subset of the default values.yaml defined in `profile-{name}.yaml` file at the component chart folder and can override only a section, or even the whole file if required. If the profile file is not present, the Kyma Operator will use default set of values from the `values.yaml` file.

Reviewer 3: ```suggestion
The profile is a subset of the default values.yaml defined in `profile-{name}.yaml` file at the component chart root folder. For example, `profile-evaluation.yaml` file defines settings for `evaluation` profile. A profile can override only a section, or even the whole file if required. If the profile file is not present, the Kyma Operator will use default set of values from the `values.yaml` file.
```

> The profile is a subset of the chart's `values.yaml` defined in `profile-{name}.yaml` file at the component chart root folder.  For example, the file `profile-evaluation.yaml` defines settings for `evaluation` profile. Values from the profile override settings from `values.yaml`. A profile can override only a section, or even the whole file if required. If the profile file is not present, the Kyma Operator will use default set of values from the `values.yaml` file.

Reviewer 2: ```suggestion
The profile is a subset of the chart's `values.yaml` defined in the `profile-{name}.yaml` file at the component chart root folder.  For example, the `profile-evaluation.yaml` file defines settings for the `evaluation` profile. Values from the profile override settings from `values.yaml`. A profile can override not only a section but also the whole file, if required. If the profile file is not present, the Kyma Operator will use the default set of values from the `values.yaml` file.
```

> Currently supported profiles are: 

Reviewer 2: I'd move the info about profiles to the first paragraph (I think that the 2nd sentence would be the best place for it). Also, it would be great to add an explanation of these profiles - how they differ and when to use them. Please add these two sentences. If you need assistance, I am happy to help 🙂

> - evaluation

Reviewer 2: ```suggestion
- Evaluation
```

> - production

Reviewer 2: ```suggestion
- Production
```

> The following fragment of the `values.yaml` shows definition limitRange settings:

Reviewer 2: ```suggestion
For example, the following fragment of the `values.yaml` file defines the **limitRange** settings:
```

```yaml
limitRange:
  max:
    memory: 4Gi
  default:
    memory: 96Mi
  defaultRequest:
    memory: 32Mi
```

> To override `limitRange.max.memory` put the following in the `profile-{name}.yaml` file: 

Reviewer 3: ```suggestion
To override `limitRange.max.memory` for `production` profile put the following in the `profile-production.yaml` file: 
```

> To override `limitRange.max.memory` for the `production` profile put the following in the `profile-production.yaml` file: 

Reviewer 2: ```suggestion
To override **limitRange.max.memory** for the `production` profile, add the following to the `profile-production.yaml` file: 
```

```yaml
limitRange:
  max:
    memory: 5Gi
```

>**NOTE:** If the `profile-{name}.yaml` file doesn't exist in a component chart you can simply create it.

> On top of the applied profile the Kyma Operator will put user-defined overrides. For more information, see the [overrides section](#configuration-helm-overrides-for-kyma-installation).

Reviewer 3: ```suggestion
>**NOTE:** User-provided overrides have precedence over the applied profile. For more information, see the [overrides section](#configuration-helm-overrides-for-kyma-installation).
```

Author: 2 notes one after another? wouldn't it look a bit strange?

Author: @klaudiagrz what do you say?

Reviewer 2: We try to avoid such situations but we also already have such cases in the documentation so I wouldn't worry about that too much 🙂

Reviewer 3: OK, it doesn't have to be a NOTE. I'll remove the emphasis to avoid double-notes.

> >**NOTE:** If the `profile-{name}.yaml` file doesn't exist in a component chart you can simply create it.

Reviewer 2: ```suggestion
>**NOTE:** If the `profile-{name}.yaml` file doesn't exist in the component chart, you can simply create it.
```

---
title: Profiles
type: Configuration
---

The Kyma Operator allows you to use pre-defined profiles to install or upgrade Kyma.

> Currently supported profiles are: 

Reviewer 2: ```suggestion
The currently supported profiles are: 
```

- Evaluation - a profile with limited resources that you can use for trial purposes

> - Production - a profile with full resource that you can use on production

Reviewer 2: ```suggestion
- Production - a profile with full resources that you can use on production
```

> The profile is a subset of the chart's `values.yaml` defined in the `profile-{name}.yaml` file at the component chart root folder.  For example, the `profile-evaluation.yaml` file defines settings for the `evaluation` profile. Values from the profile override settings from `values.yaml`. A profile can override not only a section but also the whole file, if required. If the profile file is not present, the Kyma Operator will use the default set of values from the `values.yaml` file.

Reviewer 2: ```suggestion
A profile is a subset of the chart's `values.yaml` defined in the `profile-{name}.yaml` file at the component chart root folder.  For example, the `profile-evaluation.yaml` file defines settings for the `evaluation` profile. Values from the profile override settings from `values.yaml`. A profile can override not only a section but also the whole file if necessary. If the profile file is not present, the Kyma Operator uses the default set of values from the `values.yaml` file.
```
